### ISSUE TITLE
AdguardHome DNS caching old value from /etc/hosts

### ISSUE THREAD

**Issue by kesavananbu** [AdguardTeam/AdGuardHome#7512]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Linux, ARMv7

### Installation

Docker

### Setup

On one machine

### AdGuard Home version

0.107.55

### Action

Before restarting the adguard docker, /etc/hosts has entry for 
172.23.0.5 plex

After restarting the container, i have the updated the entry in /etc/hosts 
172.23.0.11 plex

If i execute the nslookup or enter the plex in browser the ip it resolved to 172.23.0.5 instead of 172.23.0.11.
So basically it caching and not refreshing the entires for dynamic data. 

```sh
nslookup plex 192.168.1.2
```

### Expected result

Server:         192.168.1.2
Address:        192.168.1.2#53

Non-authoritative answer:
Name:   plex
Address: 172.23.0.11

### Actual result

Server:         192.168.1.2
Address:        192.168.1.2#53

Non-authoritative answer:
Name:   plex
Address: 172.23.0.5

### Additional information and/or screenshots

![upload_issue](https://github.com/user-attachments/assets/cc887085-e863-4de8-9412-79234a2ea05a)

**Comment by kesavananbu** [comment]
Similar issue #6711

**Comment by Monschichi** [comment]
It's not only returning cached data, it also don't add or remove new data. It simply don't reload /etc/hosts on change.
When I'm adding new entries to /etc/hosts, the old data from upstream DNS servers are returned.
Only a restart fixes it, even a SIGHUP, which should reload the config, assuming /etc/hosts counts as config, don't work.

**Comment by harvii** [comment]
I'm running into this issue as well on Docker (well technically Podman).

`/etc/hosts` is mounted in the container and I confirmed that changes in the original file are reflected in the container, but AGH does not detected these changes unless I restart the container.

AdGuard Home version: v0.107.55

**Comment by m4rcu5** [comment]
Looks like I might be a year late to the party, but the issue does still persist.

I discovered #6711 which mentioned there is a possible fix in edge. So I've tried that as well.
AdGuard Home version: v0.108.0-a.1241+63dc65ec

The issue is the same as described by @Monschichi and @harvii, no changes to the `/etc/hosts` file are reflected in the DNS server post-startup. Neither does a SIGHUP reload the file.

AdGuard running as a podman container on Vyos. The hosts `/etc/hosts` file is mounted as read-only in the container.
Changes are reflected correctly in the `/etc/hosts` file on the host, as well as the mounted file in the container.

I did notice that Vyos seems to replace the hostsfile on every update, aka the inode changes with every update.
